hands-on-exercise/4-run-in-prod/3-migration-prod.md

```diff
@@ -0,0 +1,616 @@
+    --project-name checkers-prod up \
+    --detach
+```
+
+In fact, at this stage, there is no difference with the previous prod setup, which was what we now call v1. Blocks are being created.
+
+Confirm you are on v1:
+
+* One way to confirm is to use checkers v2 to query Carol's node for the leaderboard:
+
+    ```sh
+    $ docker run --rm -it \
+        --network checkers-prod_net-public \
+        checkersd_i:v2-alpine \
+        query checkers show-leaderboard \
+        --node "tcp://node-carol:26657"
+    ```
+
+    It returns:
+
+    ```txt
+    Error: rpc error: code = Unknown desc = unknown query path: unknown request
+    ```
+
+    This confirms that there are no leaderboards in storage.
+
+* Another way to confirm is to see to which folder is Cosmovisor's `current` folder soft linking:
+
+    ```sh
+    $ docker exec -it node-carol \
+        ls -l /root/.checkers-upgrade/cosmovisor/current
+    ```
+
+    This should return:
+
+    ```txt
+    ... /root/.checkers-upgrade/cosmovisor/current -> /root/.checkers-upgrade/cosmovisor/genesis
+    ```
+
+    Again confirming that it is running v1, as found in `.../genesis`.
+
+## Make an upgrade proposal
+
+You will need Alice and Bob's addresses, so take them from the keyrings found on their respective _desktops_.
+
+<CodeGroup>
+
+<CodeGroupItem title="Alice">
+
+```sh
+$ alice=$(echo password | docker run --rm -i \
+    -v $(pwd)/prod-sim/desk-alice:/root/.checkers \
+    checkersd_i:v1-alpine \
+    keys \
+    --keyring-backend file --keyring-dir /root/.checkers/keys \
+    show alice --address)
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Bob">
+
+```sh
+$ bob=$(echo password | docker run --rm -i \
+    -v $(pwd)/prod-sim/desk-bob:/root/.checkers \
+    checkersd_i:v1-alpine \
+    keys \
+    --keyring-backend file --keyring-dir /root/.checkers/keys \
+    show bob --address)
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+Copying what was done in the [previous migration section](/hands-on-exercise/4-run-in-prod/2-migration.md#governance-proposal), with one block every 5 seconds, you make a proposal to run in 15 minutes, i.e. 180 blocks.
+
+Find the current block height with:
+
+```sh
+$ docker run --rm -it \
+    --network checkers-prod_net-public \
+    checkersd_i:v1-alpine status \
+    --node "tcp://node-carol:26657" \
+    | jq -r ".SyncInfo.latest_block_height"
+```
+
+This returns something like:
+
+```txt
+1000
+```
+
+With a minimum deposit of 10,000,000 upawn, you can now have Alice send the governance proposal from her desktop to Carol's public node:
+
+```sh
+$ echo password | docker run --rm -i \
+    -v $(pwd)/prod-sim/desk-alice:/root/.checkers \
+    --network checkers-prod_net-public \
+    checkersd_i:v1-alpine \
+    tx gov submit-proposal software-upgrade v1tov2 \
+    --node "tcp://node-carol:26657" \
+    --title "v1tov2" \
+    --description "Increase engagement via the use of a leaderboard" \
+    --upgrade-height 1180 \
```

Comment:
not sure if it is worth the complexity and if it makes `--upgrade-height` more clear but you could do

`--upgrade-height $(($latest_block_height + 15*60/5))`

if you export `.SyncInfo.latest_block_height`.

Reply:
Maybe no need in this case. It's not like `$alice` that we reuse.